### PR TITLE
Handle trip reversal with missing address

### LINKED ISSUE #677: Check for missing field on reverse
Check for one or both fields missing on trip reverse button click.

See #659.

### PR DESCRIPTION
**Pull request by flibbertigibbet**
Let reverse trip button work if origin, destination, or both are missing.
Fixes #677.

### CODE REVIEW

**Comment by flibbertigibbet** [comment]
Might need some additional cleanup after #668 goes in, which makes the itineraries list show/hide again.

File: src/app/scripts/cac/control/cac-control-directions.js [hunk 1]
```diff
@@ -332,11 +326,18 @@ CAC.Control.Directions = (function (_, $, moment, Control, Geocoder, Routing, Ty
...
+        // have both origin and destination
+        if (directions.origin && directions.destination) {
+            // set on this object and validate
+            setDirections('origin', [destination.feature.geometry.y, destination.feature.geometry.x]);
+            setDirections('destination', [origin.feature.geometry.y, origin.feature.geometry.x]);
+        } else if (directions.origin) {
+            setDirections('destination', [origin.feature.geometry.y, origin.feature.geometry.x]);
+        } else if (directions.destination) {
+            setDirections('origin', [destination.feature.geometry.y, destination.feature.geometry.x]);
+        }
```
Comment (KlaasH) [inline src/app/scripts/cac/control/cac-control-directions.js]: These are two independent cases now, right?  I don't see anything special happening in the "both" case.

More importantly, the typeahead `setValue` calls above are triggering events that cause `directions.origin` or `directions.destination` to get cleared.  I think these should be using the local copies of `origin` and `destination` saved at the top.

File: src/app/scripts/cac/control/cac-control-directions.js [hunk 2]
```diff
@@ -332,11 +326,18 @@ CAC.Control.Directions = (function (_, $, moment, Control, Geocoder, Routing, Ty
...
 
-        // update the directions for the reverse trip
+        // update the directions for the reverse trip, or clear itineraries
```
Comment (KlaasH) [inline src/app/scripts/cac/control/cac-control-directions.js]: I'm not seeing itineraries get cleared.  Or rather, they're disappearing from the map but not from the sidebar.

Comment (flibbertigibbet) [inline src/app/scripts/cac/control/cac-control-directions.js]: Should be fixed by #668; see note above. Code sets 'hidden' class on sidebar, but styling for 'hidden' got removed when we stripped out Bootstrap.

Comment (flibbertigibbet) [inline src/app/scripts/cac/control/cac-control-directions.js]: Just pushed rebase on redesign, so itineraries hide now.

File: src/app/scripts/cac/control/cac-control-directions.js (hunk 1 above)
Comment (flibbertigibbet) [inline src/app/scripts/cac/control/cac-control-directions.js]: I originally wrote it to only call to `planTrip` in the 'both' case, but yes, now they're independent.

Pretty sure we do want/need to `setValue`. Clearing out the `error` class for an empty field should be a special case here.

Comment (KlaasH) [inline src/app/scripts/cac/control/cac-control-directions.js]: Yeah, the `setValue` is doing necessary work, but it's also clearing the origin or destination from `directions` before we're done with it.  But the conditions can just use the local `origin` and `destination` variables, which is already what the `setDirections` calls are using.

**Comment by flibbertigibbet** [comment]
Going to hold off on further rework on this until #689 is merged

**Comment by flibbertigibbet** [comment]
Rebased and tidied a bit. This should be better behaved now. A field will go red on trip reverse if it was empty or incomplete; not sure if that's a bad thing or not.

**Comment by KlaasH** [comment]
:+1:

**Pull request merged**